File: python/102 TrafficLight.py
import threading
import time
from enum import Enum

class TrafficColor(Enum):
    RED = 9
    YELLOW = 1
    GREEN = 3

class TrafficLightThread(threading.Thread):
    def __init__(self, color):
        super().__init__()
        self.color = color

    def run(self):
        print(f"{self.color.name} active")
        time.sleep(self.color.value)
        print(f"{self.color.name} inactive")


green = TrafficLightThread(TrafficColor.GREEN)
yellow = TrafficLightThread(TrafficColor.YELLOW)
red = TrafficLightThread(TrafficColor.RED)

green.start()
green.join()

yellow.start()
yellow.join()

red.start()

File: java/102 TrafficLight.java
package code;
enum TrafficColor {
    RED(9000),
    YELLOW(1000),
    GREEN(3000);

    private final int onTimeInMillis;

    TrafficColor(int onTimeInMillis) {
        this.onTimeInMillis = onTimeInMillis;
    }

    public int getOnTimeInMillis() {
        return onTimeInMillis;
    }
}
public class Test {
	static class TrafficLightThread extends Thread {
        private final TrafficColor color;

        public TrafficLightThread(TrafficColor color) {
            this.color = color;
        }

        @Override
        public void run() {
            System.out.printf("%s active\n", color);
            try {
                Thread.sleep(color.getOnTimeInMillis());
            } catch (InterruptedException e) {
                throw new RuntimeException(e);
            }
            System.out.printf("%s inactive\n", color);
        }
    public static void main(String[] args) throws InterruptedException {
    	TrafficLightThread red = new TrafficLightThread(TrafficColor.RED);
        TrafficLightThread yellow = new TrafficLightThread(TrafficColor.YELLOW);
        TrafficLightThread green = new TrafficLightThread(TrafficColor.GREEN);

        green.start();
        green.join();
        yellow.start();
        yellow.join();
        red.start();
    }
}
}